File: src/application/contracts/IngredientRepository.ts
import { Ingredient } from '@domain/entities/Ingredient.js'
import { Id } from '@domain/valueobjects/Id.js'

export interface IngredientRepository {
  save(ingredient: Ingredient): Promise<void>
  findById(id: Id): Promise<Ingredient | null>
  findByName(name: string): Promise<Ingredient | null>
  findAll(): Promise<Ingredient[]>
  update(ingredient: Ingredient): Promise<void>
  delete(id: Id): Promise<void>
  existsById(id: Id): Promise<boolean>
  existsByName(name: string): Promise<boolean>
}

File: api/src/main/java/br/unifor/costify/application/contracts/IngredientRepository.java
package br.unifor.costify.application.contracts;

import br.unifor.costify.domain.entity.Ingredient;
import br.unifor.costify.domain.valueobject.Id;
import java.util.List;
import java.util.Optional;

public interface IngredientRepository {

  Ingredient save(Ingredient ingredient);

  Optional<Ingredient> findById(Id id);

  List<Ingredient> findAll();

  boolean existsByName(String name);

  void deleteById(Id id);
}
